include/swift/AST/TypeAttr.def

```diff
@@ -83,6 +82,7 @@ SIMPLE_SIL_TYPE_ATTR(inout, Inout)
 SIMPLE_SIL_TYPE_ATTR(inout_aliasable, InoutAliasable)
 SIMPLE_SIL_TYPE_ATTR(in_guaranteed, InGuaranteed)
 SIMPLE_SIL_TYPE_ATTR(in_constant, InConstant)
+SIMPLE_SIL_TYPE_ATTR(noescape, NoEscape)
```

Comment:
@rjmccall I don't suppose you remember why we didn't make `noescape` (and `async`) a `SIMPLE_SIL_TYPE_ATTR` in https://github.com/swiftlang/swift/commit/41cdfb04adfe13839af9e9867f775559ee307296? Mostly asking in terms of risk here - we could instead leave this change out of 6.0 and just add the missing completions in.

Reply:
I intentionally didn't put any effort in that patch into recategorizing the existing attributes.  I wanted it to be focused on the high-level representation change.

I don't know why they're not simple attributes in SIL.  They certainly look like simple attributes.

Reply:
All good, was just checking that there wasn't any underlying reason that we missed. Thanks!